R/dominantTaxa.R

```diff
@@ -124,6 +138,19 @@ setMethod("dominantTaxa", signature = c(x = "SummarizedExperiment"),
 
                    # Stores abundances
                   mat <- assay(tmp, abund_values)
+
+                  # Changes the name of species to the name of corresponding rank
+                  # If name contains upper rank, it is cut off
+                  # Tax_factors[rownames(mat)] finds corresponding name
+                  rownames(mat) <- tax_factors[rownames(mat)]
+
+                  # (strsplit(x, "_"))[[1]] splits string from underscores and
+                  # and gives a list of individual ranks
+                  # tail(x, n=1) gives the last rank from list
+                  # --> upper ranks are removed from the name
+                  rownames(mat) <- lapply(rownames(mat), function(x) tail(strsplit(x, "_")[[1]], n=1))
```

Comment:
You are assigning a list to vector. This works probably, but better style would be to use `vapply` like this:

```
vapply(strsplit(x, "_"),tail,character(1),n = 1)
```

Reply:
Changed